### ISSUE TITLE
set_state_to_close_pending match is unclear

### ISSUE THREAD

**Issue by agrover** [mozilla/neqo#605]
```
match self.state {
    ResponseStreamState::WaitingForResponseHeaders => {
        self.conn_events.header_ready(self.stream_id, None, true);
        self.state = ResponseStreamState::Closed;
    }
    ResponseStreamState::ReadingData { .. } => {}
    ResponseStreamState::WaitingForData
    | ResponseStreamState::WaitingForFinAfterTrailers => {
        self.conn_events.data_readable(self.stream_id)
    }
    _ => unreachable!("Closing an already closed transaction."),
}
```
Message seems to say  `_` case only includes `Closed`/`ClosedPending`, but that also (accidentally?) includes `DecodingHeaders` enum variant.

**Comment by ddragana** [comment]
This function is never called in DecodingHeaders state. It is intentional.

After #572 I was not very sure about the name of this function. At that time I could not think of a better name.
Maybe change_state_fin_recv. I will think about it a bit more. Maybe I move thinks a bit to make it nicer.

**Comment by agrover** [comment]
OK cool. The unreachable message was specific enough so that it really didn't apply to all unmentioned variants, which was what led to my confusion. Ironically, I think with a less detailed (worse?) message I wouldn't have even looked twice.

I'm going to just close this, it doesn't even deserve low priority.
